parsel/selector.py

```diff
@@ -537,7 +580,7 @@ def remove(self) -> None:
             )
 
         try:
-            parent.remove(self.root)
+            parent.remove(self.root)  # type: ignore[union-attr]
```

Comment:
mypy doesn't expect exceptions to be raised so it complains that `parent` can be None. I'm actually not sure why do we catch exceptions instead of doing checks here.